cli/lib/archive.ts

```diff
@@ -28,13 +32,32 @@ export async function createArchive(
 			path.join( siteFolder, 'wp-content' ),
 			'wp-content',
 			( entry: EntryData ) => {
-				if ( entry.name.includes( '.git' ) || entry.name.includes( 'node_modules' ) ) {
+				if ( shouldExcludeEntry( entry.name ) ) {
+					return false;
+				}
+				if ( entry.stats?.isSymbolicLink() ) {
 					return false;
 				}
+
 				return entry;
 			}
 		);
 
+		for ( const symlink of symlinks ) {
+			const { symbolicPath, realPath } = symlink;
+			const archivePath = path.relative( siteFolder, symbolicPath );
+			if ( symlink.isDirectory ) {
+				archive.directory( realPath, archivePath, ( entry: EntryData ) => {
+					if ( shouldExcludeEntry( entry.name ) ) {
+						return false;
+					}
+					return entry;
```

Comment:
I am not excluding symlinks in the symlinked directories to allow for potential inner symlinks. 

Additionally, in this initial version, I am not following the symlinks inside the symlinked directories to reduce complexity. If that is a common use case, it could be tackled as a follow-up.